src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/AppInsights/AppInsightsLogger.cs

```diff
@@ -0,0 +1,312 @@
+        {
+            _telemetryClient = client;
+            _categoryName = categoryName ?? DefaultCategoryName;
+            _filter = filter;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state,
+            Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string formattedMessage = string.Empty;
+            if (formatter != null)
+            {
+                formattedMessage = formatter(state, exception);
+            }
+
+            IEnumerable<KeyValuePair<string, object>> stateValues = state as IEnumerable<KeyValuePair<string, object>>;
+
+            // We only support lists of key-value pairs. Anything else we'll skip.
+            if (stateValues == null)
+            {
+                return;
+            }
+
+            // Log a function result
+            if (_categoryName == LoggingCategories.Results)
+            {
+                LogFunctionResult(stateValues, exception);
+                return;
+            }
+
+            // Log an aggregate record
+            if (_categoryName == LoggingCategories.Aggregator)
+            {
+                LogFunctionResultAggregate(stateValues);
+                return;
+            }
+
+            // Log an exception
+            if (exception != null)
+            {
+                LogException(logLevel, stateValues, exception, formattedMessage);
+                return;
+            }
+
+            // Otherwise, log a trace
+            LogTrace(logLevel, stateValues, formattedMessage);
+        }
+
+        private void LogException(LogLevel logLevel, IEnumerable<KeyValuePair<string, object>> values, Exception exception, string formattedMessage)
+        {
+            ExceptionTelemetry telemetry = new ExceptionTelemetry(exception);
+            telemetry.Message = formattedMessage;
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+            ApplyScopeProperties(telemetry);
+            ApplyCustomProperties(telemetry, logLevel, values);
+            _telemetryClient.TrackException(telemetry);
+        }
+
+        private void LogTrace(LogLevel logLevel, IEnumerable<KeyValuePair<string, object>> values, string formattedMessage)
+        {
+            TraceTelemetry telemetry = new TraceTelemetry(formattedMessage);
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+            ApplyScopeProperties(telemetry);
+            ApplyCustomProperties(telemetry, logLevel, values);
+            _telemetryClient.TrackTrace(telemetry);
+        }
+
+        private void ApplyCustomProperties(ISupportProperties telemetry, LogLevel logLevel, IEnumerable<KeyValuePair<string, object>> values)
+        {
+            telemetry.Properties.Add(LoggingKeys.CategoryName, _categoryName);
+            telemetry.Properties.Add(LoggingKeys.Level, logLevel.ToString());
```

Comment:
I think Application Insights traces and exceptions already has severity level as a separate field